### PR TITLE
Upgrading kube-prometheus-stack to v62.7.0

### PR DESCRIPTION
**Pull request by vedantjoshi84**
## ISSUE(S):
[PMK-6534](https://platform9.atlassian.net/browse/PMK-6534)

## SUMMARY:
Upgraded `kube-prometheus-stack` version from v52.1.0 (monitoring version - v0.68.0) to v62.7.0 (monitoring version - v0.76.1) 

## TESTING

* Created a helm chart
```
ubuntu@vedant-airgap:~/pf9/pf9-kube-prometheus-helm-chart$ helm package ./charts/kube-prometheus-stack --destination /home/ubuntu/deploy/
Successfully packaged chart and saved it to: /home/ubuntu/deploy/kube-prometheus-stack-62.7.0.tgz
```

* Installed helm-chart
```
root@test-pf9-qbert-bare-os-u20-3375226-631-1:~# helm install proms kube-prometheus-stack-62.7.0.tgz 
NAME: proms
LAST DEPLOYED: Fri Sep 13 10:58:53 2024
NAMESPACE: default
STATUS: deployed
REVISION: 1
NOTES:
kube-prometheus-stack has been installed. Check its status by running:
  kubectl --namespace pf9-monitoring get pods -l "release=proms"

Visit https://github.com/prometheus-operator/kube-prometheus for instructions on how to create & configure Alertmanager and Prometheus instances using the Operator.
```

```
root@test-pf9-qbert-bare-os-u20-3375226-631-1:~# helm ls 
NAME 	NAMESPACE	REVISION	UPDATED                                	STATUS  	CHART                       	APP VERSION
proms	default  	1       	2024-09-13 11:21:44.494458969 +0000 UTC	deployed	kube-prometheus-stack-62.7.0	v0.76.1
```

```
root@test-pf9-qbert-bare-os-u20-3375226-631-1:~# kubectl get svc -n pf9-monitoring
NAME                             TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)                      AGE
alertmanager-operated            ClusterIP   None            <none>        9093/TCP,9094/TCP,9094/UDP   2m19s
pf9-prom-alertmanager            ClusterIP   10.21.76.42     <none>        9093/TCP,8080/TCP            2m22s
pf9-prom-operator                ClusterIP   10.21.169.231   <none>        443/TCP                      2m22s
pf9-prom-prometheus              ClusterIP   10.21.7.52      <none>        9090/TCP,8080/TCP            2m22s
pf9-prometheus-stack-mon-svc     ClusterIP   None            <none>        8080/TCP                     2m22s
prometheus-operated              ClusterIP   None            <none>        9090/TCP                     2m18s
proms-grafana                    NodePort    10.21.182.113   <none>        80:31376/TCP                 2m22s
proms-kube-state-metrics         ClusterIP   10.21.231.96    <none>        8080/TCP                     2m22s
proms-prometheus-node-exporter   ClusterIP   10.21.199.26    <none>        9100/TCP                     2m22s
```

```
root@test-pf9-qbert-bare-os-u20-3375226-631-1:~# k get pods -n pf9-monitoring
NAME                                       READY   STATUS    RESTARTS   AGE
alertmanager-pf9-prom-alertmanager-0       2/2     Running   0          21m
monhelper-fd7d9dd95-4ss9f                  1/1     Running   0          21m
pf9-prom-operator-77d86cf9b-m46g8          1/1     Running   0          21m
prometheus-pf9-prom-prometheus-0           2/2     Running   0          21m
proms-grafana-688cd67cc6-sn7dp             3/3     Running   0          21m
proms-kube-state-metrics-b94bb8894-mrbdw   1/1     Running   0          21m
proms-prometheus-node-exporter-cmrg8       1/1     Running   0          21m
proms-prometheus-node-exporter-fv4kf       1/1     Running   0          21m
proms-prometheus-node-exporter-rpblj       1/1     Running   0          21m
```

* Checked Prometheus UI
```
kubectl port-forward  prometheus-pf9-prom-prometheus-0       9090:9090 -n pf9-monitoring
```

Screenshot 2024-09-13 at 3 51 47 PM

* Scrape targets seen

Screenshot 2024-09-13 at 3 52 14 PM
Screenshot 2024-09-13 at 3 52 50 PM

* Grafana UI:

Screenshot 2024-09-13 at 5 16 56 PM
Screenshot 2024-09-13 at 5 17 05 PM
Screenshot 2024-09-13 at 5 17 15 PM

* Tested the upgrade from v52.1.0 to v62.7.0 
* Steps: 

1. Delete the prometheus-node-exporter daemonset
```
root@test-pf9-qbert-bare-os-u20-3375226-630-2:~# k get ds -n pf9-monitoring 
NAME                             DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR            AGE
proms-prometheus-node-exporter   2         2         2       2            2           kubernetes.io/os=linux   15m
root@test-pf9-qbert-bare-os-u20-3375226-630-2:~# k delete ds proms-prometheus-node-exporter -n pf9-monitoring
daemonset.apps "proms-prometheus-node-exporter" deleted
```

2. Run these commands to update the CRDs before applying the upgrade.
```
kubectl apply --server-side -f https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/v0.76.0/example/prometheus-operator-crd/monitoring.coreos.com_alertmanagerconfigs.yaml --force-conflicts
kubectl apply --server-side -f https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/v0.76.0/example/prometheus-operator-crd/monitoring.coreos.com_alertmanagers.yaml --force-conflicts
kubectl apply --server-side -f https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/v0.76.0/example/prometheus-operator-crd/monitoring.coreos.com_podmonitors.yaml --force-conflicts
kubectl apply --server-side -f https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/v0.76.0/example/prometheus-operator-crd/monitoring.coreos.com_probes.yaml --force-conflicts
kubectl apply --server-side -f https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/v0.76.0/example/prometheus-operator-crd/monitoring.coreos.com_prometheusagents.yaml --force-conflicts
kubectl apply --server-side -f https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/v0.76.0/example/prometheus-operator-crd/monitoring.coreos.com_prometheuses.yaml --force-conflicts
kubectl apply --server-side -f https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/v0.76.0/example/prometheus-operator-crd/monitoring.coreos.com_prometheusrules.yaml --force-conflicts
kubectl apply --server-side -f https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/v0.76.0/example/prometheus-operator-crd/monitoring.coreos.com_scrapeconfigs.yaml --force-conflicts
kubectl apply --server-side -f https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/v0.76.0/example/prometheus-operator-crd/monitoring.coreos.com_servicemonitors.yaml --force-conflicts
kubectl apply --server-side -f https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/v0.76.0/example/prometheus-operator-crd/monitoring.coreos.com_thanosrulers.yaml --force-conflicts
```

3. Run `helm upgrade` 
```
root@test-pf9-qbert-bare-os-u20-3375226-630-2:~# helm list 
NAME 	NAMESPACE	REVISION	UPDATED                                	STATUS  	CHART                       	APP VERSION
proms	default  	1       	2024-09-13 11:14:03.808902998 +0000 UTC	deployed	kube-prometheus-stack-52.1.1	v0.68.0    
root@test-pf9-qbert-bare-os-u20-3375226-630-2:~# helm upgrade --install proms kube-prometheus-stack-62.7.0.tgz 


Release "proms" has been upgraded. Happy Helming!
NAME: proms
LAST DEPLOYED: Fri Sep 13 11:35:04 2024
NAMESPACE: default
STATUS: deployed
REVISION: 2
NOTES:
kube-prometheus-stack has been installed. Check its status by running:
  kubectl --namespace pf9-monitoring get pods -l "release=proms"

Visit https://github.com/prometheus-operator/kube-prometheus for instructions on how to create & configure Alertmanager and Prometheus instances using the Operator.
```

```
Croot@test-pf9-qbert-bare-os-u20-3375226-630-2:~# k get svc -n pf9-monitoring
NAME                             TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)                      AGE
alertmanager-operated            ClusterIP   None            <none>        9093/TCP,9094/TCP,9094/UDP   38m
pf9-prom-alertmanager            ClusterIP   10.21.180.72    <none>        9093/TCP,8080/TCP            38m
pf9-prom-operator                ClusterIP   10.21.102.197   <none>        443/TCP                      38m
pf9-prom-prometheus              ClusterIP   10.21.191.207   <none>        9090/TCP,8080/TCP            38m
pf9-prometheus-stack-mon-svc     ClusterIP   None            <none>        8080/TCP                     38m
prometheus-operated              ClusterIP   None            <none>        9090/TCP                     38m
proms-grafana                    NodePort    10.21.146.3     <none>        80:30337/TCP                 38m
proms-kube-state-metrics         ClusterIP   10.21.83.139    <none>        8080/TCP                     38m
proms-prometheus-node-exporter   ClusterIP   10.21.83.199    <none>        9100/TCP                     38m
```

```root@test-pf9-qbert-bare-os-u20-3375226-630-2:~# kubectl get pods -n pf9-monitoring
NAME                                       READY   STATUS    RESTARTS   AGE
alertmanager-pf9-prom-alertmanager-0       2/2     Running   0          38s
monhelper-fd7d9dd95-klb4v                  1/1     Running   0          22m
pf9-prom-operator-77d86cf9b-8d7dh          1/1     Running   0          60s
prometheus-pf9-prom-prometheus-0           2/2     Running   0          30s
proms-grafana-688cd67cc6-xrbpw             3/3     Running   0          60s
proms-kube-state-metrics-b94bb8894-brklq   1/1     Running   0          60s
proms-prometheus-node-exporter-vhkgw       1/1     Running   0          60s
proms-prometheus-node-exporter-w9jd4       1/1     Running   0          60s
```

4. Verified the Prometheus and Grafana UI working and accessible. 
For the Grafana UI, the username is `admin` and password is `prom-operator` which is given in the `values.yaml` file. 

[PMK-6534]: https://platform9.atlassian.net/browse/PMK-6534?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by mithilarun — DISMISSED** [review]

File: charts/kube-prometheus-stack/Chart.lock [hunk 1]
```diff
@@ -4,15 +4,15 @@ dependencies:
...
-  version: 5.14.0
+  version: 5.25.1
 - name: prometheus-node-exporter
   repository: https://prometheus-community.github.io/helm-charts
-  version: 4.23.2
+  version: 4.39.0
 - name: grafana
   repository: https://grafana.github.io/helm-charts
-  version: 6.60.6
+  version: 8.5.1
```
Comment (cruizen) [inline charts/kube-prometheus-stack/Chart.lock:13]: This also upgrade Grafana to v8.5.1 https://github.com/grafana/helm-charts/releases/tag/grafana-8.5.1 
I am unable to tag Jayanth here somehow. Can you ask him to experiment with this version?

**Review by cruizen — DISMISSED** [review]
/lgtm

File: charts/kube-prometheus-stack/values.yaml [hunk 2]
```diff
@@ -1200,6 +1200,7 @@ grafana:
   ##
   service:
     portName: http-web
+    type: NodePort
```
Comment (cruizen) [inline charts/kube-prometheus-stack/values.yaml:1203]: Is this required?

**Review by cruizen — DISMISSED** [review]
/lgtm

**Review by cruizen — APPROVED** [review]
/lgtm

**Review by srinivaschavan — APPROVED** [review]

**Review by srm6867 — APPROVED** [review]
/lgtm

**Pull request merged**